pkg/controllers/jobset_controller.go

```diff
@@ -874,21 +837,31 @@ func findFirstFailedJob(failedJobs []*batchv1.Job) *batchv1.Job {
 	return firstFailedJob
 }
 
-// findJobFailureTime is a helper function which extracts the Job failure time from a Job,
+// findJobFailureTimeAndReason is a helper function which extracts the Job failure condition from a Job,
 // if the JobFailed condition exists and is true.
-func findJobFailureTime(job *batchv1.Job) *metav1.Time {
+func findJobFailureCondition(job *batchv1.Job) *batchv1.JobCondition {
 	if job == nil {
 		return nil
 	}
 	for _, c := range job.Status.Conditions {
 		// If this Job failed before the oldest known Job failiure, update the first failed job.
 		if c.Type == batchv1.JobFailed && c.Status == corev1.ConditionTrue {
-			return &c.LastTransitionTime
+			return &c
 		}
 	}
 	return nil
 }
 
+// findJobFailureTime is a helper function which extracts the Job failure time from a Job,
+// if the JobFailed condition exists and is true.
+func findJobFailureTime(job *batchv1.Job) *metav1.Time {
```

Comment:
if this and the function above are only used in failure policy handling, then I suggest to move them to failure_policy.go file.

Reply:
This is done. I moved the functions along with a few others.